### PR TITLE
Add proxy support in deployment script

### PR DESCRIPTION
**Pull request by liguangbo**
1. Some users need to set the proxy to download file.
2. When deploying,  we need download easy-rsa. Otherwise, we can't make a CA to authenticate users. 
3. The sudo command can't transfer the environment variables.
4.  We provide some parameters in config-default.sh to solve this problem. Across these parameters, we can set proxy before the download, and unset proxy after download. Fix this problem.

### CODE REVIEW

**Comment by liguangbo** [comment]
@jszczepkowski Could please review this PR, Thanks.

File: cluster/ubuntu/config-default.sh [hunk 1]
```diff
@@ -66,3 +66,8 @@ ENABLE_CLUSTER_UI="${KUBE_ENABLE_CLUSTER_UI:-true}"
 
 # Optional: Enable setting flags for kube-apiserver to turn on behavior in active-dev
 #RUNTIME_CONFIG=""
+
+# Optional: Enable proxy in kube-up. If you neet set proxy when download easy-rsa in kube-up.
+HTTP_PROXY=""
```
Comment (jszczepkowski) [inline cluster/ubuntu/config-default.sh]: How this value is going to be set? I think it is better to take it from env. variable:
HTTP_PROXY="${HTTP_PROXY:-}"

**Comment by jszczepkowski** [comment]
This PR is bit hacky as it is using http_proxy and https_proxy environment variables instead of adding new arguments (containing http_proxy) to make-ca-cert.sh script. The new arguments to make-ca-cert.sh will also make the extremely hacky modifications of /etc/sudoers unnecessary.

In my opinion, we should change make-ca-cert.sh script to accept http-proxy (if it is needed).

@fgrzadkowski What do you think?

@liguangbo Please wait with applying my comments for @fgrzadkowski opinion (he is responsible for cluster lifecycle).

**Comment by liguangbo** [comment]
@jszczepkowski How about add the content of make-ca-cert.sh into util.sh. We can download the binaries and easy-rsa at the same time in build.sh. Then copy the easy-rsa to master and make cert. So we can avoid this problem in deploying.

**Comment by jszczepkowski** [comment]
> How about add the content of make-ca-cert.sh into util.sh. 
If this is copy-pasting of the code, this is rather not a good idea.

**Comment by liguangbo** [comment]
@jszczepkowski As /kubernetes/cluster/gce/util.sh function create-cert do, only create cert we need.

File: cluster/ubuntu/util.sh [hunk 2]
```diff
@@ -457,3 +461,24 @@ function kube-push {
 function prepare-e2e() {
   echo "Ubuntu doesn't need special preparations for e2e tests" 1>&2
 }
+
+# Set proxy environment variables
+function setProxy() {
+  source "kube/${KUBE_CONFIG_FILE-"config-default.sh"}"
```
Comment (fgrzadkowski) [inline cluster/ubuntu/util.sh]: You shouldn't need to source it here. All functions in this file depends on variables in config-default.sh.

File: cluster/ubuntu/util.sh [hunk 3]
```diff
@@ -457,3 +461,24 @@ function kube-push {
...
+  if grep '!env_reset' /etc/sudoers;then
```
Comment (fgrzadkowski) [inline cluster/ubuntu/util.sh]: How about using env_keep instead? It's more targeted.

File: cluster/ubuntu/util.sh [hunk 4]
```diff
@@ -457,3 +461,24 @@ function kube-push {
...
+  else
+    sed -i -e 's|env_reset|!env_reset|' /etc/sudoers
+  fi
+  export http_proxy=${HTTP_PROXY}
+  export https_proxy=${HTTPS_PROXY}
+}
+
+# Unset proxy environment variables
+function unsetProxy() {
+  if grep '!env_reset' /etc/sudoers;then
```
Comment (fgrzadkowski) [inline cluster/ubuntu/util.sh]: This function does not revert ```env_reset``` in ```/etc/sudoers```, i.e. if it was ```!env_reset``` before ```setProxy()```, this function will set it to ```env_reset```.

**Comment by liguangbo** [comment]
@fgrzadkowski I fix it as you said. How is the setproxy now.Thank you.

**Comment by liguangbo** [comment]
@fgrzadkowski I send a new PR #13964 to solve this problem in non-intrusive way.

**Pull request closed without merging**